File: csharp/high-scores/HighScores.cs
using System;
using System.Collections.Generic;
using System.Linq;

public class HighScores
{
    IList<int> scores;

    public HighScores(List<int> scores)
    {
        this.scores = scores;
    }

    public IList<int> Scores() => scores;

    public int Latest() => scores.Last();

    public int PersonalBest() => scores.Max();

    public IList<int> PersonalTopThree() => scores.OrderByDescending(x => x).Take(3).ToList();
}

File: java/high-scores/src/main/java/HighScores.java
import java.util.Comparator;
import java.util.List;

class HighScores {
  List<Integer> highScores;

  public HighScores(List<Integer> highScores) {
    this.highScores = highScores;
  }

  List<Integer> scores() {
    return highScores;
  }

  Integer latest() {
    return highScores.get(highScores.size() - 1);
  }

  Integer personalBest() {
    return highScores.stream().max(Comparator.naturalOrder()).get();
  }

  List<Integer> personalTopThree() {
    return highScores.stream().sorted(Comparator.reverseOrder()).limit(3).toList();
  }
}
